arch/arm/src/nrf52/nrf52_sdc.c

```diff
@@ -0,0 +1,609 @@
+ * Name: rng_handler
+ ****************************************************************************/
+
+static void rng_handler(void)
+{
+  sdc_RNG_IRQHandler();
+}
+
+/****************************************************************************
+ * Name: power_clock_isr
+ ****************************************************************************/
+
+static int power_clock_isr(int irq, FAR void *context, FAR void *arg)
+{
+  MPSL_IRQ_CLOCK_Handler();
+
+  return 0;
+}
+
+/****************************************************************************
+ * Name: rtc0_handler
+ ****************************************************************************/
+
+static void rtc0_handler(void)
+{
+  MPSL_IRQ_RTC0_Handler();
+}
+
+/****************************************************************************
+ * Name: timer0_handler
+ ****************************************************************************/
+
+static void timer0_handler(void)
+{
+  MPSL_IRQ_TIMER0_Handler();
+}
+
+/****************************************************************************
+ * Name: radio_handler
+ ****************************************************************************/
+
+static void radio_handler(void)
+{
+  MPSL_IRQ_RADIO_Handler();
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+int nrf52_sdc_initialize(void)
+{
+  int ret;
+  int32_t required_memory;
+  sdc_cfg_t cfg;
+
+  /* Initialize device data */
+
+  memset(&g_sdc_dev, 0, sizeof(g_sdc_dev));
+  nxsem_init(&g_sdc_dev.exclsem, 0, 1);
+
+  /* Register interrupt handler for normal-priority events. SWI5 will be
+   * used by MPSL to delegate low-priority work
+   */
+
+  irq_attach(NRF52_IRQ_SWI5_EGU5, swi_isr, NULL);
+  irq_attach(NRF52_IRQ_POWER_CLOCK, power_clock_isr, NULL);
+
+  up_enable_irq(NRF52_IRQ_SWI5_EGU5);
+  up_enable_irq(NRF52_IRQ_POWER_CLOCK);
+
+  up_prioritize_irq(NRF52_IRQ_SWI5_EGU5, NVIC_SYSH_PRIORITY_DEFAULT);
+  up_prioritize_irq(NRF52_IRQ_POWER_CLOCK, NVIC_SYSH_PRIORITY_DEFAULT);
+
+  /* Use zero-latency interrupt for RNG as we're expected to not add any
+   * processing to the ISR
+   */
+
+  arm_ramvec_attach(NRF52_IRQ_RNG, rng_handler);
+  up_prioritize_irq(NRF52_IRQ_RNG, NVIC_SYSH_MAXNORMAL_PRIORITY);
+  up_enable_irq(NRF52_IRQ_RNG);
+
+  /* Register high-priority interrupts for specific peripherals */
+
+  arm_ramvec_attach(NRF52_IRQ_RTC0, rtc0_handler);
+  arm_ramvec_attach(NRF52_IRQ_TIMER0, timer0_handler);
+  arm_ramvec_attach(NRF52_IRQ_RADIO, radio_handler);
+
+  up_prioritize_irq(NRF52_IRQ_RTC0, MPSL_HIGH_IRQ_PRIORITY);
+  up_prioritize_irq(NRF52_IRQ_TIMER0, MPSL_HIGH_IRQ_PRIORITY);
+  up_prioritize_irq(NRF52_IRQ_RADIO, MPSL_HIGH_IRQ_PRIORITY);
+
+  up_enable_irq(NRF52_IRQ_RTC0);
+  up_enable_irq(NRF52_IRQ_TIMER0);
+  up_enable_irq(NRF52_IRQ_RADIO);
+
+  /* TODO: how do WFI again after high priority interrupt wakes MCU up? */
```

Comment:
Not sure what you are asking here.  It if is a the WFI in up_idle and there is nothing READY then it will just continue the idle loop and hit WFI again.

Reply:
Right, I'll remove the comment.